src/main/scala/dpla/ingestion3/enrichments/SpatialEnrichment.scala

```diff
@@ -0,0 +1,275 @@
+  * constructor when you create a spatial enrichment.
+  *
+  * @example
+  *          object Geocoder extends Twofisher {
+  *            override def hostname = {
+  *              System.getenv("GEOCODER_HOST") match {
+  *                case h if h.isInstanceOf[String] => h
+  *                case _ => "localhost"
+  *              }
+  *            }
+  *          }
+  *
+  * @see SpatialEnrichmentIntegrationTest
+  */
+trait Twofisher {
+
+  def hostname: String = "localhost"
+  def port: String = "8081"
+
+  /**
+    * Return the JValue object from an HTTP request to Twofishes.
+    *
+    * @param query The term that will be queried: coordinates or name
+    * @return      The response JValue, containing geographic interpretations
+    */
+  def geocoderResponse(query: String): JValue = {
+    val baseURI = s"http://$hostname:$port/query"
+    val response: HttpResponse[String] =
+      Http(baseURI)
+        .param("lang", "en")
+        .param("responseIncludes", "PARENTS,DISPLAY_NAME")
+        .param("query", query)
+        .asString
+    parse(response.body)
+  }
+}
+
+
+/**
+  * The DplaPlace spatial object enrichment
+  *
+  * @param geocoder  A singleton object representing your geocoder
+  */
+class SpatialEnrichment(geocoder: Twofisher) {
+
+  implicit val formats = DefaultFormats // Formats for json4s
+  val logger: Logger = Logger.getLogger(getClass.getName)
+
+  /**
+    * Given a Place that needs amending, Return the query term string that will
+    * be passed to Twofishes, depending on whether the coordinates or name are
+    * already filled in.
+    *
+    * Return nothing if the place is the United States. We don't want to resolve
+    * "United States" unless it's something like "United States--California"
+    * with two dashes, because we'd get too many results otherwise that are at
+    * the geographic center of the U.S., in Kansas.  Note that we'll skip
+    * lookups on all countries in enrich(), but this saves some processing time,
+    * avoiding unnecessary lookups in the first place, because of the volume of
+    * Places that we expect to be given as "United States."
+    *
+    * @param place  The edm:Place object to consider
+    * @return       The query term: either the name or coordinates, or None
+    */
+  private def queryTerm(place: DplaPlace): Option[String] = {
+    if (place.coordinates.isDefined) place.coordinates
+    else if (place.name.isDefined) {
+      val name = place.name.mkString
+      name match {
+        case s if s matches """^United States(?!\-\-)""" => None
+        case s if s matches "^USA" => None
+        case _ => place.name
+      }
+    }
+    else None
+  }
+
+  /**
+    * Return the place's display name, for sourceResource.spatial.name
+    *
+    * @param j  The JValue object for the Twofishes response body
+    * @return   The place's display name
+    */
+  private def displayName(j: JValue): Option[String] = {
+    Option(((j \ "interpretations")(0) \ "feature")
```

Comment:
It wasn't intuitive at first, and I think I got to this by looking at an example somewhere else. It looks like the return value has to be explicitly wrapped in this Option wrapper to match the declared return type.